commcare_connect/opportunity/tasks.py

```diff
@@ -54,3 +54,13 @@ def generate_visit_export(opportunity_id: int, date_range: str, status: list[str
     export_tmp_name = f"{now().isoformat()}_{opportunity.name}_visit_export.{export_format}"
     storages["default"].save(export_tmp_name, ContentFile(content))
     return export_tmp_name
+
+
+@celery_app.task()
+def generate_payment_export(opportunity_id: int, export_format: str):
+    opportunity = Opportunity.objects.get(id=opportunity_id)
+    dataset = export_empty_payment_table(opportunity)
+    content = dataset.export("xlsx")
+    export_tmp_name = f"{now().isoformat()}_{opportunity.name}_payment_export.{export_format}"
```

Comment:
Will this lead to the file having different content-encoding (xlsx) and file format incase a user selects `csv`?

Reply:
good catch, I thought I updated all those instances